sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs

```diff
@@ -672,6 +672,16 @@ public partial class ShareFileLease
         public string LeaseId { get { throw null; } }
         public int? LeaseTime { get { throw null; } }
     }
+    public partial class ShareFileOpenReadConcurrentModificationException : System.Exception, System.Runtime.Serialization.ISerializable
```

Comment:
This name is a little too long.  I think we should be careful about including `OpenRead` in case you add `OpenWrite`.  I'd also worry a little about calling ending this with `ConcurrentModificationException` when we're not deriving from that class as is the typical convention in .NET.  We could maybe drop `ShareFile` since the `ResourceUri` makes it nonspecific, but I like including `Share` somewhere in there in case we ever need another one of these.

What if we went with `ShareFileModificationException` since the property they pass in is `AllowModifications`?

Reply:
If we drop `Share` then this exception should go to `Azure.Storage.Common`. Given knowledge we have I'd rather keep it in shares for now and extract common type into commons later if needed.